flake.nix

```
@@ -28,6 +28,21 @@
 
         crane-nightly = crane-lib.overrideToolchain rust-nightly;
 
+        check-crate = pkgs.writeShellApplication rec {
+          name = "check-crate";
+          runtimeInputs = [ rust-nightly ];
+          text = ''
+            EXTRA_FEATURES=""
+            if [[ -n "''${2-}" ]]; then
```

What’s with the `''`? Shouldn’t this be:

```suggestion
            if [[ -n "${2-}" ]]; then
```

---

' ' is escape character to avoid conflict with nix host.